tensorflow/compiler/xla/client/xla_builder.h

```diff
@@ -314,6 +324,10 @@ class XlaBuilder {
     ShapeIndex param_index;
   };
 
+  // Looks up the HloInstruction and sets the frontend attribute "attribute" to
+  // "value".
+  Status AddFrontendAttribute(const XlaOp& op, string attribute, string value);
```

Comment:
Why do we need both this and `SetFrontendAttributes` etc.?

Reply:
Updating the doc.

Also I was thinking about renaming it `SetInstructionFrontendAttribute()` as it only sets the attribute attached to the instruction and not the builder's one.